# kmerKernels
Phylogenetic tree reconstruction requires construction of a multiple sequence alignment (MSA) from sequences. However, alignment can be quite challenging for very large genomes. An interesting recent approach uses intensive hashing method to count the fraction of all unique k-mers appearing in either genome that are common to both to estimate the evolutionary distance, which avoids the need of assembly and alignment. 

In this project, I try to implement this k-mer idea using a more space efficient method based on the 2BWT. The program is written to process raw data, build 2BWT, count distinct and shared k-mers and generate the distance table. Using the simulated reads data from paper, I am able to construct the similar common k-mer table and phylogeny tree as  the one in AAF paper. 

To further apply the method on large data set, 4 real genomes are tested and the resulting phylogeny tree looks the same as the one generated by the program phyloT.
